Liberty server won't start when specifying java security manager in Java 24+

When trying to start a Liberty server with the security manager enabled in the bootstrap.properties file, like this
```
websphere.java.security
```

The Liberty server fails to start with messages similar to this in the console.log:
```
WARNING: package sun.security.action not in java.base
Launching com.ibm.ws.security.csiv2.fat.servlet.client.java2 (WebSphere Application Server 25.0.0.1/wlp-1.0.97.202501102305) on OpenJDK 64-Bit Server VM, version 24-ea+31-3600 (en_US)
CWWKE0005E: The runtime environment could not be launched.
CWWKE0018E: An exception occurred while launching the runtime environment: java.lang.UnsupportedOperationException: Setting a system-wide Policy object is not supported
com.ibm.ws.kernel.boot.LaunchException: Caught unexpected exception java.lang.UnsupportedOperationException: Setting a system-wide Policy object is not supported
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.rethrowException(KernelBootstrap.java:714)
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.go(KernelBootstrap.java:221)
	at com.ibm.ws.kernel.boot.Launcher.handleActions(Launcher.java:243)
	at com.ibm.ws.kernel.boot.Launcher.createPlatform(Launcher.java:119)
	at com.ibm.ws.kernel.boot.cmdline.EnvCheck.main(EnvCheck.java:61)
	at com.ibm.ws.kernel.boot.cmdline.EnvCheck.main(EnvCheck.java:37)
Caused by: java.lang.UnsupportedOperationException: Setting a system-wide Policy object is not supported
	at java.base/java.security.Policy.setPolicy(Policy.java:114)
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.enableJava2SecurityIfSet(KernelBootstrap.java:458)
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.buildClassLoader(KernelBootstrap.java:412)
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.go(KernelBootstrap.java:196)
	... 4 more
```

But according to our [design documentation for handling the disabling of the security manager in Java 18](https://github.com/OpenLiberty/open-liberty/issues/20289), we are supposed to ignore the security manager request, post a message and start the server.

--- a/dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/KernelBootstrap.java
+++ b/dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/KernelBootstrap.java
@@ -305,7 +305,7 @@ public class KernelBootstrap {
             return serverLock.waitForStop();
         }
 
-        // Server did not propertly start (no delegate), so stop is fine.
+        // Server did not properly start (no delegate), so stop is fine.
         return ReturnCode.OK;
     }
 
@@ -438,7 +438,7 @@ public class KernelBootstrap {
      * Set Java 2 Security if enabled
      */
     public static void enableJava2SecurityIfSet(BootstrapConfig bootProps, List<URL> urlList) {
-        if (bootProps.get(BootstrapConstants.JAVA_2_SECURITY_PROPERTY) != null) {
+        if (bootProps.get(BootstrapConstants.JAVA_2_SECURITY_PROPERTY) != null && javaVersion() < 24) {
 
             NameBasedLocalBundleRepository repo = new NameBasedLocalBundleRepository(bootProps.getInstallRoot());
             File bestMatchFile = repo.selectBundle("com.ibm.ws.org.eclipse.equinox.region",
@@ -493,7 +493,7 @@ public class KernelBootstrap {
         String bsConsoleFormat = bootProps.get("com.ibm.ws.logging.console.format");
         String envConsoleFormat = System.getenv("WLP_LOGGING_CONSOLE_FORMAT");
 
-        //boostrap format should take precedence
+        //bootstrap format should take precedence
         String consoleFormat = bsConsoleFormat != null ? bsConsoleFormat : envConsoleFormat;
 
         if (productDisplayName == null) {
@@ -755,5 +755,15 @@ public class KernelBootstrap {
             return true;
         }
         return false;
+    }
+
+    private static int javaVersion() {
+        String version = System.getProperty("java.version");
+        String[] versionElements = version.split("\\D"); // split on non-digits
+
+        // Pre-JDK 9 the java.version is 1.MAJOR.MINOR
+        // Post-JDK 9 the java.version is MAJOR.MINOR
+        int i = Integer.valueOf(versionElements[0]) == 1 ? 1 : 0;
+        return Integer.valueOf(versionElements[i]);
     }
 }

